dev/provider_packages/prepare_provider_packages.py

```diff
@@ -1133,6 +1133,14 @@ def get_package_pip_name(provider_package_id: str, backport_packages: bool):
         return f"apache-airflow-providers-{provider_package_id.replace('.', '-')}"
 
 
+def get_provider_info(provider_package_id: str):
+    provider_yaml_file_name = os.path.join(get_source_package_path(provider_package_id), "provider.yaml")
+    if not os.path.exists(provider_yaml_file_name):
+        raise Exception(f"The provider.yaml file is missing: {provider_yaml_file_name}")
+    with open(provider_yaml_file_name) as provider_file:
+        return provider_file.read()
```

Comment:
```suggestion
        return yaml.safe_load(provider_file)
```

Reply:
Yeah. I was going to try that and see if it works in all cases

Reply:
Seems it does even in most complex providers.